gobblin-dataset-config/src/main/java/gobblin/dataset/config/ConfigStore.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (C) 2014-2015 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.dataset.config;
+
+import java.util.List;
+import java.util.Map;
+
+import com.typesafe.config.Config;
+
+/**
+ * A ConfigStore is the configuration store per scheme.
+ * @author mitu
+ *
+ */
+public interface ConfigStore {
+
+  /**
+   * @return the latest configuration version in this ConfigStore
+   */
+  public String getLatestVersion();
+
+  /**
+   * @return the configuration store scheme, example DAI-ETL, DALI or Espresso
+   */
+  public String getScheme();
+  
+  /**
+   * Loaded the latest configuration version. 
+   */
+  public void loadConfigs();
+  
+  /**
+   * Loaded the specified configuration version.
+   * @param version ,the configuration version to be loaded
+   */
+  public void loadConfigs(String version);
+  
+  /**
+   * @param urn input urn
```

Comment:
Please be specific in the javadocs what "input urn" means, e.g. "a dataset urn", "a tag urn", or both.